### PR TITLE
Add alternative bitcoin clients

### PR DESCRIPTION
**Pull request by JustinAiken**
# Description

This adds casks for [Bitcoin Classic](https://bitcoinclassic.com/) and [Bitcoin Unlimited](https://www.bitcoinunlimited.info) clients to the existing Bitcoin Core and Bitcoin XT clients, and makes it so all 4 bitcoin-related casks are aware of conflicts with the other 4.

# Checklist
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.
  (Rubocop is freaking out on me locally)
- [ ] The commit message includes the cask’s name and version.
  - See questions
- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

# Questions

- Would you rather it be three commits like this - one for each of the new casks, one for the conflicts? Or squash some/all of them?
- Do you want versions for each of them in the commit msg, being new?
- Anything else?

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Three commits are preferred, but thee PRs are even more appropriate. More often than not this type of pull request brings problems with one or more of them, and it halts the inclusion of the other ones.

File: Casks/bitcoin-classic.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  name     'Bitcoin Classic'
+  homepage 'https://bitcoinclassic.com/'
+
+  conflicts_with cask: 'bitcoin-core',
+                 cask: 'bitcoin-unlimited',
+                 cask: 'bitcoin-xt'
+
+  depends_on     macos: '>= :mountain_lion'
+
+  app 'Bitcoin-Qt.app', target: 'Bitcoin Classic.app'
```
Comment (vitorgalvao) [inline Casks/bitcoin-classic.rb]: [`target` requires a comment](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/app.md#target-should-only-be-used-in-select-cases).

File: Casks/bitcoin-unlimited.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  name     'Bitcoin Unlimited'
+  homepage 'https://www.bitcoinunlimited.info'
+
+  conflicts_with cask: 'bitcoin-classic',
+                 cask: 'bitcoin-core',
+                 cask: 'bitcoin-xt'
+
+  depends_on     macos: '>= :mountain_lion'
+
+  app 'Bitcoin-Qt.app', target: 'Bitcoin Unlimited.app'
```
Comment (vitorgalvao) [inline Casks/bitcoin-unlimited.rb]: [`target` requires a comment](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/app.md#target-should-only-be-used-in-select-cases).

File: Casks/bitcoin-classic.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  preflight do
+    set_permissions "#{staged_path}/Bitcoin-Qt.app", '0755'
+  end
+
+  # Bitcoin classic names it preferences file same as Bitcoin Core:
```
Comment (vitorgalvao) [inline Casks/bitcoin-classic.rb]: Remove this one.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by JustinAiken** [comment]
> Three commits are preferred, but thee PRs are even more appropriate

Okay! I'll close this, and do 3 separates (including the changes requested here)

**Pull request closed without merging**